### DISCUSSION TITLE
I want to get a randomly specified number of elements from the result returned by the Keys command, what command should I use?

### DISCUSSION THREAD

**Discussion by fcgxz2003** [redis/go-redis#2638] [Q&A]
Keys command needs to traverse the entire redis, can it traverse the specified number and return?

**Comment by elena-kolevska** [comment]
Hi @fcgxz2003 . `KEYS` goes through the whole keyspace on the server side. We can't control it on the client. It's also important to note that it is a *blocking* command, meaning no other requests can be served while `KEYS` is running.

From the docs:
"Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases. This command is intended for debugging and special operations, such as changing your keyspace layout. Don't use KEYS in your regular application code. If you're looking for a way to find keys in a subset of your keyspace, consider using [SCAN](https://redis.io/commands/scan) or [sets](https://redis.io/topics/data-types#sets)."

I would suggest looking into the `SCAN` command.
